src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -208,33 +208,33 @@ public virtual async Task<ValidationContext> FullValidationAsync(ChainedHeader h
         }
 
         /// <inheritdoc/>
-        public virtual async Task<ValidationContext> PartialValidationAsync(ChainedHeader header, Block block)
+        public virtual ValidationContext PartialValidationAsync(ChainedHeader header, Block block)
         {
             Guard.NotNull(header, nameof(header));
             Guard.NotNull(block, nameof(block));
 
             var validationContext = new ValidationContext { BlockToValidate = block, ChainedHeaderToValidate = header };
             RuleContext ruleContext = this.CreateRuleContext(validationContext);
 
-            await this.ExecuteRulesAsync(this.partialValidationRules, ruleContext).ConfigureAwait(false);
+            this.ExecuteRulesAsync(this.partialValidationRules, ruleContext);
 
             if (validationContext.Error != null)
                 this.HandleConsensusError(validationContext);
 
             return validationContext;
         }
 
-        private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules, RuleContext ruleContext)
+        private void ExecuteRulesAsync(IEnumerable<ConsensusRuleBase> asyncRules, RuleContext ruleContext)
```

Comment:
If we were going to do this... remove Async.

Reply:
ah right yes I missed that one, I will do it in another PR (as its only cosmetics and we'll need to run all tests again for such a small change)